src/frequenz/channels/util/_select.py

```diff
@@ -87,24 +54,20 @@ class Select:
         ```
     """
 
-    def __init__(self, **kwargs: Receiver[Any]) -> None:
+    def __init__(self, *receivers: Receiver[Any]) -> None:
         """Create a `Select` instance.
 
         Args:
-            **kwargs: sequence of receivers
+            *receivers: A set of receivers to select from.
         """
-        self._receivers = kwargs
+        self._receivers: Dict[str, Receiver[Any]] = {
+            f"0x{id(r):x}": r for r in receivers
```

Comment:
it might be nice to be consistent with our use of `id` or `hash`.  If we want to use `id` here,  then the iterator should return `list`, for which the `in` operator uses `id` to compare (through `eq` -> `is` -> `id`).   If we want to return a `set` from the iterator, which uses `hash` in the `in` operator,  we should use `hash` here too.

My preference is to use `id` here and return a `list`, because that is explicit.

Reply:
Mmm, I only used `id()` here for debugging purposes. I use the key as the task name and also when you `print(instance)` it gives you the `id`, not the hash (if `str` is not customized). I didn't thought about `id` vs `hash` at all.

I see your point about consistency and mixing `id` with `hash` when using `in` with the results of the iterator, but using `hash` will make things harder to debug and in practice there shouldn't be any differences, `id` and `hash` should both work to identify an instance uniquely (assuming receivers won't override `hash` in a way that they expect 2 different Receiver instances produce the same hash on purpose).

I'd rather not return a `list` in the iterator because a `set` makes it clear that there are no duplicates. With `set` is like we are encoding the `unique` property into the *type system*.  Also returning a `list` means we need a linear search over the `ready_set` for the `in` operator (I know this is only a conceptual annoyance as with small lists, which is probably what we'll have, a linear search should be fast enough anyway).

If you are still concerned about the `id`/`hash` inconsistency, I would rather use `hash()` for the key/task name and sacrifice some debuggability instead of having less ideal API typing.

Reply:
why do you say we'll sacrifice readability if we use hash?

``` python
>>> hash(object())
8759965091979
>>> id(object())
140159441471728
```

they both look similarly difficult to read, for me. :-)

Reply:
Not readability, debuggability, because if you do `print(object)` or `print(f"this is my Object: {object}")` you will get the `id()` (if `__str__()` is not override) and not `hash()`, but then when looking at the `_ready_set` keys you'll get the hash.

It is a minor thing, this is why I say I prefer to go with `hash()` as key/task name if you really think the inconsistency is an issue.